File: python/linear_search.py
def ls_nr(arr,k,n):

	ind=-1
	for i in range(n):
		if(arr[i]==k):
			ind=i;
			break;

	return ind;


def ls_r(arr,k,i,n):
	if(arr[i]==k):
		return i;
	else:
		if(i==n-1):
			return -1;
		else:
			return ls_r(arr,k,i+1,n);
	

print("Enter the value of n:")
n=int(input())
li=[]

print("Enter "+str(n)+" elements:")
for i in range(n):
	li.append(int(input()))

print("Enter the element to be searched:")
k=int(input())
print("Index of element to be searched (non recursive)= "+str(ls_nr(li,k,n)))
print("Index of element to be searched (recursive)= "+str(ls_r(li,k,0,n)))

File: java/linear_search.java
import java.util.*;

class linear_search{

	public static int ls_nr(int arr[],int k,int n){

		int i,ind=-1;
		for(i=0;i<n;i++){
		if(arr[i]==k){
		ind=i;
		break;
		}
		}

		return ind;
	}

	public static int ls_r(int arr[],int k,int i,int n){
		if(arr[i]==k)
			return i;
		else{
			if(i==n-1){
			
				return -1;
			}
			else
				return ls_r(arr,k,i+1,n);
		}
	}

	public static void main(String args[]){
		Scanner sc=new Scanner(System.in);

		System.out.println("Enter the number of elements:\n");
		int n=sc.nextInt();
		System.out.println("Enter "+n+" elements:\n");
		int arr[]=new int[n];
		for(int i=0;i<n;i++){
			arr[i]=sc.nextInt();
		}
		System.out.println("Enter the element to be searchedt:\n");
		int k=sc.nextInt();
		
		System.out.println("\nIndex of element to be searched (non recursive)= "+ls_nr(arr,k,n));
		System.out.println("\nIndex of element to be searched (recursive)= "+ls_r(arr,k,0,n));	
	}
}